### ISSUE TITLE
"PARAMETERS method is not supported!" when using cross-method parameters

### ISSUE THREAD

**Issue by richardklose** [seriousme/fastify-openapi-glue#52]
According to [OpenAPI 3](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#path-item-object) the following definition should be valid:
```yaml
...
paths:
  /users/{id}:
    parameters:
      # definition of parameters for all methods of this path, e.g.
      - name: id
        in: path
        required: true
    get:
      # definition of a get request
      ...
    post:
      # definition of a post request
      ...
```

But when the `parameters` key of a Path Object is used, it seems to be handled like the `get`, `post`, etc. keys, so it is used as a definition of an additional method here.

```
Error: PARAMETERS method is not supported!\n                     
    at Object.route (/path/to/project/node_modules/fastify/lib/route.js:118:15)\n                     
     at Object._route [as route] (/path/to/project/node_modules/fastify/fastify.js:162:27)\n                     
     at config.routes.forEach.item (/path/to/project/node_modules/fastify-openapi-glue/index.js:80:22)\n                     
     at Array.forEach (<anonymous>)\n                     
     at generateRoutes (/path/to/project/node_modules/fastify-openapi-glue/index.js:67:19)\n                     
     at Plugin.exec (/path/to/project/node_modules/avvio/plugin.js:89:17)\n                     
     at Boot.loadPlugin (/path/to/project/node_modules/avvio/plugin.js:175:10)\n                     
     at release (/path/to/project/node_modules/fastq/queue.js:127:16)\n                     
     at Object.resume (/path/to/project/node_modules/fastq/queue.js:61:7)\n                     
     at Plugin.finish (/path/to/project/node_modules/avvio/plugin.js:166:10)
```

Instead, this `parameters` object should be merged into all method definitions.

**Comment by seriousme** [comment]
Correct, looks like we need to add  some more logic to [parserv3](https://github.com/seriousme/fastify-openapi-glue/blob/master/lib/parser.v3.js#L135) as the spec is imho rather messy here mixing operationObjects with other stuff :-(

I'll look into it when I have some time.

Kind regards,
Hans

**Comment by seriousme** [comment]
Ok, fixed in #54 and published on npmjs.com as version 1.2.0

Kind regards,
Hans

**Comment by seriousme** [comment]
Can we close this issue ?

Kind regards,
Hans
